### PR TITLE
created maxtext readme file

### PR DESCRIPTION
**Pull request by kocchop**

### CODE REVIEW

File: rosetta/rosetta/projects/maxtext/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+Once the container is up and running, you can quickly launch a job with the following command
+```
+python3 MaxText/train.py MaxText/configs/base.yml hardware=gpu run_name=$YOUR_JOB_NAME
+```
+You can similarly launch a llama2-7b training job with following command:
+```
+python3 MaxText/train.py MaxText/configs/base.yml run_name=$YOUR_JOB_NAME use_iota_embed=true scan_layers=true\
+    steps=15 per_device_batch_size=2 model_name=llama2-7b remat_policy=minimal_flash enable_checkpointing=false logits_dot_in_fp32=false\
+    base_output_directory=local_train dataset_path=local dataset_type=synthetic attention=dot_product\
+    max_target_length=4096 hardware=gpu
```
Comment (terrykong) [inline rosetta/rosetta/projects/maxtext/README.md:36]: nit: Could you change it to one arg per line? It makes reading/updating a little easier.
```suggestion
python3 MaxText/train.py \
    MaxText/configs/base.yml \
    run_name=$YOUR_JOB_NAME \
    use_iota_embed=true scan_layers=true \
    steps=15 \
    per_device_batch_size=2 \
    model_name=llama2-7b \
    remat_policy=minimal_flash enable_checkpointing=false \
    logits_dot_in_fp32=false \
    base_output_directory=local_train \
    dataset_path=local \
    dataset_type=synthetic attention=dot_product \
    max_target_length=4096 \
    hardware=gpu
```

Comment (terrykong) [reply rosetta/rosetta/projects/maxtext/README.md:36]: Also, does this use multiprocess by default?

File: rosetta/rosetta/projects/maxtext/scripts/example_slurm.sub [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+RUN_NAME=${RUN_NAME:-"demo"}
+
+read -r -d '' cmd <<EOF
+echo "*******STARTING********" \
+&& cd $MAXTEXT_DIR \
+&& nvidia-smi \
+&& python3 MaxText/train.py MaxText/configs/base.yml run_name=$RUN_NAME use_iota_embed=true scan_layers=false\
+    steps=15 per_device_batch_size=2 model_name=$MODEL remat_policy=minimal_flash enable_checkpointing=false logits_dot_in_fp32=false\
+    base_output_directory=local_train dataset_path=local dataset_type=synthetic attention=cudnn_flash_te\
+    max_target_length=4096 hardware=gpu_multiprocess
```
Comment (terrykong) [inline rosetta/rosetta/projects/maxtext/scripts/example_slurm.sub:96]: Similar nit to this multiline command:
```suggestion
&& python3 MaxText/train.py \
    MaxText/configs/base.yml \
    run_name=$RUN_NAME \
    use_iota_embed=true scan_layers=false \
    steps=15 \
    per_device_batch_size=2 \
    model_name=$MODEL \
    remat_policy=minimal_flash \
    enable_checkpointing=false logits_dot_in_fp32=false \
    base_output_directory=local_train \
    dataset_path=local \
    dataset_type=synthetic attention=cudnn_flash_te \
    max_target_length=4096 \
    hardware=gpu_multiprocess
```

File: rosetta/rosetta/projects/maxtext/scripts/example_slurm.sub [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+# create run specific output directory for ease of analysis
+mkdir -p "${BASE_WORKSPACE_DIR}/${OUTPUT_DIR}/${RUN_NAME}"
+
+# redirect both stdout and stderr in the same file for ease of analysis
+OUTFILE="${BASE_WORKSPACE_DIR}/${OUTPUT_DIR}/${RUN_NAME}/output-%j-%n-%t.txt"
+
+echo $cmd
+srun -o $OUTFILE -e $OUTFILE --container-image="$CONTAINER" $MOUNTS $EXPORTS bash -c "${cmd}"
+set +x
```
Comment (terrykong) [inline rosetta/rosetta/projects/maxtext/scripts/example_slurm.sub:107]: You can drop this `set -x` since it's not needed
```suggestion
```

**Review by terrykong — CHANGES_REQUESTED** [review]

File: rosetta/rosetta/projects/maxtext/README.md (hunk 1 above)
Comment (kocchop) [reply rosetta/rosetta/projects/maxtext/README.md:36]: No, the interactive run does not use multiprocess

**Review by terrykong — APPROVED** [review]

**Pull request merged**